File: 1239-maximum-length-of-a-concatenated-string-with-unique-characters/1239-maximum-length-of-a-concatenated-string-with-unique-characters.cs
public class Solution {
        int res = 0;
        public int MaxLength(IList<string> arr)
        {
            var ls = arr.ToHashSet().ToList();
            dfs(ls, "",0);
            return res;
        }
        public void dfs(List<string> ls,string temp,int st)
        {
            res = Math.Max(res, temp.Length);
            for (int i = st; i < ls.Count; i++)
            {
                var s = $"{temp}{ls[i]}";
                if (!check(s))
                {
                    continue;
                }
                else
                {
                    dfs(ls, s, i + 1);
                }
            }
        }
        public bool check(string a)
        {
            var f = new int[26];
            for (int i = 0; i < a.Length; i++)
            {
                if(f[a[i] - 'a']++ > 0)
                {
                    return false;
                }
            }
            return true;
        }
}

File: 1239-maximum-length-of-a-concatenated-string-with-unique-characters/1239-maximum-length-of-a-concatenated-string-with-unique-characters.cpp
class Solution {
public:
    bool check(string s){
        unordered_set<char> p(s.begin(),s.end());
        if(p.size()!=s.size()) return false;
        return true;
    }
    int h(vector<string>& arr,int ind,string s){
        if(!check(s)){
            return 0;
        }
        int res=s.size();
        for (int i = ind; i < arr.size(); i++)
        {
            res=max(res,h(arr,i+1,s+arr[i]));
        }
        return res;
    }
    int maxLength(vector<string>& arr) {
        return h(arr,0,"");
    }
};